react/components/LocaleSwitcher/index.js

```diff
@@ -0,0 +1,92 @@
+import classnames from 'classnames'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import IconGlobe from '../icon/Globe'
+import IconCaretDown from '../icon/CaretDown'
+
+const findLocale = (currentLocale, availableLocales) => {
+  const localeObj = availableLocales.find(
+    ({ id }) => id.split('-')[0] === currentLocale.split('-')[0]
+  )
+  return localeObj || availableLocales[0]
+}
+
+class LocaleSwitcher extends Component {
+  static propTypes = {
+    availableLocales: PropTypes.arrayOf(PropTypes.shape({
+      text: PropTypes.string,
+      id: PropTypes.string,
+    })),
+    currentLocale: PropTypes.string,
+    switchLocale: PropTypes.func,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      opened: false,
+      selectedLocale: findLocale(props.currentLocale, props.availableLocales),
+    }
+  }
+
+  handleButtonClick = () => {
+    this.setState(({ opened }) => ({ opened: !opened }))
+  }
+
+  handleLocaleClick = ({ target: { id } }) => {
+    const [, locale] = id.split('@')
```

Comment:
We decided that we are not using this code so I won't respond to all of the review, but this is a curiosity, so I'll answer.
This is [js destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) but, since the first argument is not defined, the first element of the array goes nowhere. It's the same as
`[a, locale] = id.split('@')` and never using `a` again.

It should be `@`, btw. I'm not that sure why now, but I remember having checked it. If we were to launch, I'd check again.